tensordict/utils.py

```diff
@@ -398,17 +398,32 @@ def expand_right(
 }
 
 
-def _nested_key_type_check(key: NestedKey) -> None:
-    msg = "Expected key to be a string or non-empty tuple of strings, but found {}."
-    if type(key) is str:
-        return
-    is_tuple = type(key) is tuple
-    if not is_tuple:
-        raise TypeError(msg.format(type(key)))
-    else:
-        for subkey in key:
-            if type(subkey) is not str:
-                raise TypeError(msg.format(type(subkey)))
+def is_nested_key(key: NestedKey) -> bool:
+    """Returns True if key is a NestedKey."""
+    if key is None:
+        return False
```

Comment:
Is this needed? `None` will fail the next two if statements and so return `False` anyway.

Reply:
I guess it's to catch it early since we may want to do 
```
Module(lambda: torch.rand(3), in_keys=None, out_keys=["rand"])
```

Reply:
good catch! remainder of previous version